### ISSUE TITLE
opentelemetry-otlp: '/v1/traces' is not appended to endpoint for http exporter

### ISSUE THREAD

**Issue by bouk** [open-telemetry/opentelemetry-rust#997]
The HTTP exporter doesn't add `/v1/traces` to the 'endpoint' specified in the config:

https://github.com/open-telemetry/opentelemetry-rust/blob/592e77a8f323fe5c490de9eab458b7e40e9c437b/opentelemetry-otlp/src/span.rs#L404-L419

This seems like an oversight because this is done in other places:

https://github.com/open-telemetry/opentelemetry-rust/blob/592e77a8f323fe5c490de9eab458b7e40e9c437b/opentelemetry-otlp/src/span.rs#L319-L328

https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-otlp/src/metric.rs#L235-L245

**Comment by djc** [comment]
Thanks for reporting, would you be able to submit a PR to improve on this?

**Comment by jaymell** [comment]
I believe this issue was fixed by #1210? As a result, #1056 can probably be closed as well. 

I just went on a deep dive looking to see whether this had been addressed in later versions of the crate :sweat_smile:. Since it has, it might be good to marked it as fixed.

**Comment by svix-james** [comment]
Actually to be clear, while this was fixed, we've actually found it to have a significant bug :sob::

```
    format!("{provided_or_default_endpoint}{signal_endpoint_path}")
        .parse()
        .map_err(From::from)
```

 `/v1/traces` is just added directly to `provided_or_default_endpoint`, which, if it originated from a `Url` or `Uri`, is going to have a trailing slash on it, resulting in something like `http://example.com//v1/traces`, which in most cases will be a 404.

One simple approach here is to strip off the path from `provided_or_default_endpoint` and re-build the Uri with `signal_endpoint_path` as the path instead. This would break anybody who supplies a path with the endpoint, but I think that wouldn't comply with spec anyway.

**Comment by jaymell** [comment]
Actually to be clear, while this was fixed, we've actually found it to have a significant bug 😭:

    format!("{provided_or_default_endpoint}{signal_endpoint_path}")
        .parse()
        .map_err(From::from)

/v1/traces is just added directly to provided_or_default_endpoint, which, if it originated from a Url or Uri, is going to have a trailing slash on it, resulting in something like http://example.com//v1/traces, which in most cases will be a 404.

One simple approach here is to strip off the path from provided_or_default_endpoint and re-build the Uri with signal_endpoint_path as the path instead. This would break anybody who supplies a path with the endpoint, but I think that wouldn't comply with spec anyway.
